### PR TITLE
ta: trusted_keys: Fix to align with strict buffer checks

### PR DESCRIPTION
**Pull request by b49020**
Commit e12c9f67d12c ("core: strict buffer check in syscalls following GP 1.1")
has switched to stricter buffer checks to reside in TA private memory.
So accordingly fix buffer allocations corresponding to TEE_GenerateRandom
and TEE_AEDecryptFinal APIs.

Signed-off-by: Sumit Garg <sumit.garg@linaro.org>

### CODE REVIEW

File: ta/trusted_keys/entry.c [hunk 1]
```diff
@@ -42,7 +44,15 @@ static TEE_Result get_random(uint32_t types, TEE_Param params[TEE_NUM_PARAMS])
 	if (!params[0].memref.buffer || !params[0].memref.size)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	TEE_GenerateRandom(params[0].memref.buffer, params[0].memref.size);
+	rng_buf = TEE_Malloc(params[0].memref.size, TEE_MALLOC_FILL_ZERO);
```
Comment (jforissier) [inline ta/trusted_keys/entry.c:47]: Should the buffer be tested first? In case the caller is a TA or a PTA, the additional allocation may be unnecessary.

**Review by jforissier — COMMENTED** [review]
>```
>ta: trusted_keys: Fix to align with strict buffer checks
>
>Commit e12c9f67d12c ("core: strict buffer check in syscalls following GP 1.1")
>```
Line too long (checkpatch)
>```
>has switched to stricter buffer checks to reside in TA private memory.
>So accordingly fix buffer allocations corresponding to TEE_GenerateRandom
>and TEE_AEDecryptFinal APIs.
>```
`TEE_GenerateRandom() and TEE_AEDecryptFinal() APIs.`
>```
>
>Signed-off-by: Sumit Garg <sumit.garg@linaro.org>
>```

File: ta/trusted_keys/entry.c [hunk 2]
```diff
@@ -122,10 +136,12 @@ static TEE_Result huk_ae_decrypt(TEE_OperationHandle crypto_op, uint8_t *in,
 	if (res)
 		return res;
 
+	memcpy(tag, hdr->tag, TAG_SIZE);
 	res = TEE_AEDecryptFinal(crypto_op, hdr->enc_key, enc_key_len, out,
-				 out_sz, hdr->tag, TAG_SIZE);
+				 out_sz, tag, TAG_SIZE);
 	if (res)
 		res = TEE_ERROR_SECURITY;
+	memzero_explicit(tag, TAG_SIZE);
```
Comment (jenswi-linaro) [inline ta/trusted_keys/entry.c:144]: Why do we need to clear this?

File: ta/trusted_keys/entry.c [hunk 3]
```diff
@@ -87,11 +97,14 @@ static TEE_Result huk_ae_encrypt(TEE_OperationHandle crypto_op, uint8_t *in,
...
 	struct tk_blob_hdr *hdr = (struct tk_blob_hdr *)out;
+	uint8_t iv[IV_SIZE] = { 0 };
 	uint32_t enc_key_len = in_sz;
 	uint32_t tag_len = TAG_SIZE;
 
 	hdr->reserved = 0;
-	TEE_GenerateRandom(hdr->iv, IV_SIZE);
+	TEE_GenerateRandom(iv, IV_SIZE);
+	memcpy(hdr->iv, iv, IV_SIZE);
+	memzero_explicit(iv, IV_SIZE);
```
Comment (jenswi-linaro) [inline ta/trusted_keys/entry.c:107]: Why is this cleared?

File: ta/trusted_keys/entry.c (hunk 1 above)
Comment (b49020) [reply ta/trusted_keys/entry.c:47]: Current client for this TA is Linux kernel only. IMO, at this point extra checking would be redundant.

File: ta/trusted_keys/entry.c (hunk 3 above)
Comment (b49020) [reply ta/trusted_keys/entry.c:107]: IV isn't considered sensitive, will drop it.

File: ta/trusted_keys/entry.c (hunk 2 above)
Comment (b49020) [reply ta/trusted_keys/entry.c:144]: Will drop it.

**Comment by b49020** [comment]
> > ```
> > ta: trusted_keys: Fix to align with strict buffer checks
> > 
> > Commit e12c9f67d12c ("core: strict buffer check in syscalls following GP 1.1")
> > ```
> 
> 
> Line too long (checkpatch)
> 
> 

If I wrap it then I get below error:

```
ERROR: Please use git commit description style 'commit <12+ chars of sha1> ("<title line>")' - ie: 'commit e12c9f67d12c ("core: strict buffer check in syscalls following GP 1.1")'
#6: 
Commit: e12c9f67d12c ("core: strict buffer check in syscalls following
```

**Comment by jforissier** [comment]
> > > ```
> > > ta: trusted_keys: Fix to align with strict buffer checks
> > > 
> > > Commit e12c9f67d12c ("core: strict buffer check in syscalls following GP 1.1")
> > > ```
> > 
> > 
> > Line too long (checkpatch)
> 
> If I wrap it then I get below error:
> 
> ```
> ERROR: Please use git commit description style 'commit <12+ chars of sha1> ("<title line>")' - ie: 'commit e12c9f67d12c ("core: strict buffer check in syscalls following GP 1.1")'
> #6: 
> Commit: e12c9f67d12c ("core: strict buffer check in syscalls following
> ```

Weird, works for me. The colon character maybe? It is not in the expected format.

File: ta/trusted_keys/entry.c (hunk 1 above)
Comment (jforissier) [reply ta/trusted_keys/entry.c:47]: OK makes sense.

**Comment by b49020** [comment]
> Weird, works for me. The colon character maybe? It is not in the expected format.

Ah, yes. Fixed now, thanks.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Review by jforissier — COMMENTED** [review]
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

**Comment by b49020** [comment]
Thanks, tags applied.

**Pull request merged**